doc/model.rst

```diff
@@ -0,0 +1,2354 @@
+   SmartRedis :ref:`Client functions<client_prefix_func>`.
+2. **Retrieving Data from the Orchestrator**: Users can instruct a ``Client`` to prepend a
+   ``Model`` name to a key during data retrieval, polling, or check for existence on the ``Orchestrator``
+   through SmartRedis :ref:`Client functions<client_prefix_func>`.
+
+For example, assume you have two ``Models`` in an ``Experiment``, named `model_0` and `model_1`. In each
+application code you use the function ``Client.put_tensor("tensor_0", data)`` to send a tensor named `"tensor_0"`
+to the same ``Orchestrator``. With ``Model`` key prefixing turned on, the `model_0` and `model_1`
+applications can access their respective tensor `"tensor_0"` by name without overwriting or accessing
+the other ``Models`` `"tensor_0"` tensor. In this scenario, the two tensors placed in the
+``Orchestrator`` are `model_0.tensor_0` and `model_1.tensor_0`.
+
+Enabling and Disabling
+======================
+SmartSim provides support for toggling prefixing on a ``Model`` for tensors, ``Datasets``,
+lists, ML models, and scripts. Prefixing functions from the SmartSim :ref:`Model API<model_api>` and SmartRedis :ref:`Client API<smartredis-api>` rely on
+each other to fully support SmartSim key prefixing. For example, to use the ``Client`` prefixing
+functions, a user must enable prefixing on the ``Model`` through ``Model.enable_key_prefixing()``.
+This function enables prefixing for tensors, ``Datasets`` and lists placed in an ``Orchestrator``
+by the ``Model``. This configuration can be toggled within the ``Model`` application through
+``Client`` functions, such as disabling tensor prefixing via ``Client.use_tensor_ensemble_prefix(False)``.
+
+The interaction between the prefix SmartSim `Model functions<model_prefix_func>` and SmartRedis
+`Client functions<client_prefix_func>` are documentation below.
+
+.. _model_prefix_func:
+---------------
+Model functions
+---------------
+A ``Model`` object supports two prefixing functions: ``Model.enable_key_prefixing()`` and
+``Model.register_incoming_entity()``.
+
+To enable prefixing on a ``Model``, users must use the ``Model.enable_key_prefixing()``
+function in the ``Experiment`` driver script. The key components of this function include:
+
+- Activates prefixing for tensors, ``Datasets``, and lists sent to a ``Orchestrator`` from within
+  the ``Model`` application.
+- Enables access to prefixing ``Client`` functions within the ``Model`` application. This excludes
+  the ``Client.set_data_source()`` function, where ``enable_key_prefixing()`` is not require access.
+
+.. note::
+    ML model and script prefixing is not automatically enabled through ``Model.enable_key_prefixing()``
+    and rather must be enabled within the ``Model`` application using ``Client.use_model_ensemble_prefix()``.
+
+To enable a SmartRedis ``Client`` to search a number of ``Model`` prefixes when requesting data from the ``Orchestrator``,
```

Comment:
I think this sentence and below sub bullets make things a little confusing because it gives very specific examples and not a lot of context for what is happening.  I almost feel like we don't need these bullets because we have enumerated all of these cases in the examples.  I think we could just say something as simple as 

"To enable a SmartRedis ``Client`` within a ``Model`` to be able to access data, ML models, and scripts that were set by a ``Model``, including itself, with key prefixing enabled, ``Model.register_incoming_entity()`` must be invoked.  The function  ``Model.register_incoming_entity()`` is called in the SmartSim driver script on the ``Model`` instance that will access prefixed data and the proper ``Model`` entities are provided to the function as input parameters.  The examples illustrate the utility of ``Model.register_incoming_entity()`` for all of the ``Orchestrator`` data structures.